root/usr/sbin/ns8-join

```diff
@@ -150,10 +161,69 @@ if account_provider_config['isAD'] == '1':
             print("no_available_node_for_samba_provider", file=sys.stderr)
             sys.exit(1)
     except json.JSONDecodeError:
-        pass # Ignore missing nsdc/IpAddress prop
+        # Ignore missing nsdc/IpAddress prop, assuming the provider is
+        # external.
+        account_provider_external = "1"
+        # Ensure the external AD domain is already configured in NS8:
+        for domain in domains['data']['output']['domains']:
+            if domain.get("schema", "") != "ad":
+                continue # ignore non-AD domains
+            elif domain["name"] == account_provider_domain:
+                break # Domain match found.
+        else:
+            # No domain match found.
+            print("ns8-join: a matching external AD domain was not found in NS8.", file=sys.stderr)
+            sys.exit(1)
     except Exception as e:
         print("Error:", e, file=sys.stderr)
         sys.exit(1)
+elif account_provider_config['isLdap'] == '1':
+    if '127.0.0.1' in account_provider_config['LdapURI']:
+        # We have a local OpenLDAP account provider. Acquire the domain
+        # name for migration from UI.
+        if not args.user_domain:
+            print("ns8-join: user_domain is required for OpenLDAP account provider", file=sys.stderr)
+            sys.exit(1)
```

Comment:
Not really sure that this check is safe enough.
Why just not expose a warning inside the interface but still allow the migration?